src/main/java/br/com/caelum/vraptor/test/requestflow/UserFlow.java

```diff
@@ -114,11 +109,26 @@ public UserFlow to(final String url, final HttpMethod httpMethod, final Paramete
 			@Override
 			public VRaptorTestResult call(HttpSession session) {
 				LOG.debug("starting request to " + url);
-				request = new MockHttpServletRequest(context, httpMethod.toString(), url) {
+				final String localUrl;
+				final String query;
+				if(url.contains("?")){
+					String[] parts = url.split("\\?");
+					localUrl = parts[0];
+					query = parts[1];
+				} else {
+					localUrl = url;
+					query = "";
+				}
```

Comment:
You are Welcome!
I created method splitUrl, because the param url must be "/testQuery" and not "testQuery?somerField=someValue" to work else it answer with 404.